ibis/expr/types/structs.py

```diff
@@ -26,55 +26,36 @@ def struct(
 ) -> StructValue:
     """Create a struct expression.
 
-    If the input expressions are all column expressions, then the output will be
-    a `StructColumn`.
-
-    If the input expressions are Python literals, then the output will be a
-    `StructScalar`.
+    If any of the inputs are Columns, then the output will be a `StructColumn`.
+    Otherwise, the output will be a `StructScalar`.
 
     Parameters
     ----------
     value
-        The underlying data for literal struct value or a pairs of field names
-        and column expressions.
+        Either a {`str`: `Value`} mapping,
+        or an iterable of tuples of the form (`str`, `Value`).
     type
         An instance of `ibis.expr.datatypes.DataType` or a string indicating
         the ibis type of `value`. This is only used if all of the input values
-        are literals.
+        are python literals. eg "struct<a: float, b: string>".
 
     Returns
     -------
     StructValue
-        An expression representing a literal or column struct (compound type with
-        fields of fixed types)
+        An StructScalar or StructColumn expression.
 
     Examples
     --------
-    Create a struct literal from a [](`dict`) with the type inferred
+    Create a struct scalar literal from a `dict` with the type inferred
     >>> import ibis
-    >>> t = ibis.struct(dict(a=1, b="foo"))
-
-    Create a struct literal from a [](`dict`) with a specified type
-    >>> t = ibis.struct(dict(a=1, b="foo"), type="struct<a: float, b: string>")
-
-    Specify a specific type for the struct literal
-    >>> t = ibis.struct(dict(a=1, b=40), type="struct<a: float, b: int32>")
-
-    Create a struct array from multiple arrays
     >>> ibis.options.interactive = True
-    >>> t = ibis.memtable({"a": [1, 2, 3], "b": ["foo", "bar", "baz"]})
-    >>> ibis.struct([("a", t.a), ("b", t.b)])
-    ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
-    ┃ StructColumn()              ┃
-    ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
-    │ struct<a: int64, b: string> │
-    ├─────────────────────────────┤
-    │ {'a': 1, 'b': 'foo'}        │
-    │ {'a': 2, 'b': 'bar'}        │
-    │ {'a': 3, 'b': 'baz'}        │
-    └─────────────────────────────┘
+    >>> ibis.struct(dict(a=1, b="foo"))
+
+    Specify a specific type (note the 1 is now a float):
+    >>> ibis.struct(dict(a=1, b="foo"), type="struct<a: float, b: string>")
 
-    Create a struct array from columns and literals
+    Create a struct column from a column and a scalar literal
+    >>> t = ibis.memtable({"a": [1, 2, 3]})
```

Comment:
```suggestion
    Create a struct column from a column and a scalar literal
    
    >>> t = ibis.memtable({"a": [1, 2, 3]})
```

Line breaks are required between prose and code.